packages/core/src/external/carequality/ihe-gateway-v2/ihe-gateway-v2-async.ts

```diff
@@ -28,16 +31,26 @@ export class IHEGatewayV2Async extends IHEGatewayV2 {
     cxId: string;
   }): Promise<void> {
     const lambdaClient = makeLambdaClient(Config.getAWSRegion());
-    const params = { patientId, cxId, pdRequestGatewayV2 };
-    // intentionally not waiting
-    lambdaClient
-      .invoke({
-        FunctionName: iheGatewayV2OutboundPatientDiscoveryLambdaName,
-        InvocationType: "Event",
-        Payload: JSON.stringify(params),
-      })
-      .promise()
-      .catch(processAsyncError("Failed to invoke iheGatewayV2 lambda for patient discovery"));
+    const { gateways, ...rest } = pdRequestGatewayV2;
+    const gatewayChunks =
+      gateways.length > MAX_GATEWAYS_BEFORE_CHUNK
+        ? chunk(gateways, Math.ceil(gateways.length / 2))
+        : [gateways];
+
+    for (const chunk of gatewayChunks) {
+      const newPdRequestGatewayV2 = { ...rest, gateways: chunk };
+      const params = { ...newPdRequestGatewayV2, patientId, cxId };
+
+      // intentionally not waiting
+      lambdaClient
+        .invoke({
+          FunctionName: iheGatewayV2OutboundPatientDiscoveryLambdaName,
+          InvocationType: "Event",
+          Payload: JSON.stringify(params),
+        })
+        .promise()
+        .catch(processAsyncError("Failed to invoke iheGatewayV2 lambda for patient discovery"));
+    }
```

Comment:
We're invoking the lambda multiple times, which is nice. But what about the response, don't we need a way to coordinate those in order to know when the PD is completed?